app/editor_ui/mixins.py

```diff
@@ -139,3 +140,68 @@ def dispatch(self, request, *args, **kwargs):
             )
 
         return super().dispatch(request, *args, **kwargs)
+
+
+class BreadCrumbsMixin:
+    """
+    A mixin to help with the display of breadcrumbs via the TNA Breadcrumbs component
+
+    Usage:
+    - Add BreadCrumbsMixin to your view class
+    - Set the `breadcrumb` property in the view class
+
+    Note:
+    - The implicit assumption is that all view urls are suffixed with '/'. All urls for views must have the '/' suffix,
+    if the BreadCrumbsMixin is to work.
+    """
+
+    breadcrumb = "Error"
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context["breadcrumbs"] = self._breadcrumb_calculator()
+        return context
+
+    def _breadcrumb_calculator(self):
+        """
+        Calculate the breadcrumbs from the url, from root to the current path.
+
+        Only views that have the BreadCrumbsMixin and self.breadcrumbs
+        will be added to the list of breadcrumbs.
+        """
+        parts = self.request.path.split("/")
+
+        breadcrumbs = []
+
+        parsed_url = ""
+
+        for url_part in parts[:-1]:
+            parsed_url += f"{url_part}/"
```

Comment:
`resolve(url)` will only work if the url in the urlpatterns has a `/` suffix. 

We can move away from that convention, but we'll need to then ensure that all urls do not end with a `/` suffix in `urlpatterns`.